### PR TITLE
Stop sending keep alives once the BPUP transport is closed

### PR DESCRIPTION
**Pull request by bdraco**
- Fixes a memory leak when the Home Assistant integration is reloaded

- See https://github.com/home-assistant/core/pull/57799

- Fixes https://github.com/home-assistant/core/issues/53530

### CODE REVIEW

File: bond_api/bond.py [hunk 1]
```diff
@@ -160,6 +168,8 @@ def connection_made(self, transport):
 
     def send_keep_alive(self):
         """Send a keep alive every 60 seconds per the protocol."""
+        if not self.transport:
```
Comment (pryste-sqlapi) [inline bond_api/bond.py:171]: if this is important, could you add a unit test for it?

**Review by pryste-sqlapi — APPROVED** [review]
I approved, but could you please look at my comment and check if this can be done? @bdraco

**Comment by bdraco** [comment]
@prystupa cc @pryste-sqlapi  

Coverage added

**Comment by bdraco** [comment]
Retested with HASS as well

**Pull request merged**